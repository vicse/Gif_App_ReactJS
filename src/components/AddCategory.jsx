import { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target: { value } }) => {
        setInputValue(value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const input = inputValue.trim();
        if (input.length <= 1) return;

        // setCategories((cat) => [ inputValue, ...cat ]);   
        onNewCategory(input);
        setInputValue(''); 
    }

    return (
        <form onSubmit={ onSubmit } aria-label="form">
            <input
                text="text"
                placeholder="Buscar Gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}