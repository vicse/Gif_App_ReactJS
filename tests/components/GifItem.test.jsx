import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GitItem />', () => {

    const title = 'Satoru Gojo';
    const url = 'https://jujutsukaisen.com/gojo.jpg';

    test('debe de hacer match con el snapshot', () => {

        const { container } = render(<GifItem title={ title } url={ url } /> );
        expect(container).toMatchSnapshot();

    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {

        render(<GifItem title={ title } url={ url } /> );
        // screen.debug();
        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);

    });

    test('debe de mostrar el titulo en el componente', () => {
        
        render(<GifItem title={ title } url={ url } /> );
        expect(screen.getByText(title)).toBeTruthy();
    })



})