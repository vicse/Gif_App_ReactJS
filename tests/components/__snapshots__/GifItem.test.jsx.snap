// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GitItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Satoru Gojo"
      src="https://jujutsukaisen.com/gojo.jpg"
    />
    <p>
      Satoru Gojo
    </p>
  </div>
</div>
`;
